server/provider/networkProvider.go

```diff
@@ -119,11 +124,23 @@ func (provider *networkProvider) GetBlockchainName() string {
 }
 
 // GetNativeCurrency gets the native currency (EGLD, 18 decimals)
-func (provider *networkProvider) GetNativeCurrency() resources.NativeCurrency {
-	return resources.NativeCurrency{
-		Symbol:   provider.nativeCurrencySymbol,
-		Decimals: int32(nativeCurrencyNumDecimals),
-	}
+func (provider *networkProvider) GetNativeCurrency() resources.Currency {
```

Comment:
since these methods only call the methods on provider.currenciesProvider, why not keep the currenciesProvider unexported but make its methods exported? The networkProvider could have then extend the currenciesProvider struct. To prevent code coupling, you could have use a private interface instead of *currenciesProvider pointer.

Reply:
Indeed, refactored :+1: